chainerx_cc/chainerx/routines/math.cc

```diff
@@ -1176,4 +1176,28 @@ Array IsFinite(const Array& x) {
     return out;
 }
 
+Array Where(const Array& condition, const Array& x, const Array& y) {
+    if (condition.dtype() != Dtype::kBool) {
+        throw DtypeError{"condition should be an array of dtype bool."};
+    }
+
+    if (x.dtype() != y.dtype()) {
+        throw DtypeError{"dtype of x and y should be same."};
+    }
+
+    const Array& x_b = x.BroadcastTo(condition.shape());
+    const Array& y_b = y.BroadcastTo(condition.shape());
```

Comment:
How about broadcasting to the following shape?
```cpp
Shape out_shape = internal::BroadcastShapes(condition.shape(), internal::BroadcastShapes(x.shape(), y.shape()));
```

Reply:
Done.